sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -526,6 +525,21 @@ public Flux<OpenConnectionResponse> openConnectionsAndInitCaches(String containe
         return Flux.empty();
     }
 
+    public void prepareRequestForAuth(RxDocumentServiceRequest request, String resourceName) {
```

Comment:
this method will never be called twice for the same request right? 
else it will keep encoding the encoded resourceAddress

I do not think it will happen as even for each retry, it is a new request

Reply:
Correct - also it will get the resourcename from upstream caller - so there the requirement to not double-encode.

Reply:
Which is in RxDocumentClientImpl where we ensure this right now